File: 02 백준/10 문자열/Java/src/Prob1316.java

/**
 * title  : 1316. 그룹 단어 체커 (Silver 5)
 * time   : 124ms
 * memory : 14176KB
 */

 import java.io.*;
 import java.util.*;
 
 public class Prob1316 {
    
    public static void main(String[] args) throws NumberFormatException, IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        
        int answer = 0;
        for (int n = 0; n < N; n++) {

            String word = br.readLine();
            int length = word.length();

            Set<Character> check = new HashSet<>();

            int idx = 0;
            boolean flag = true;
            while (idx < length) {

                char temp = word.charAt(idx);

                if (check.contains(temp)) {
                    flag = false;
                    break;
                }

                check.add(temp);
                while (idx < length && word.charAt(idx) == temp) {
                    idx++;
                }
            }

            if (flag) {
                answer += 1;
            }
        }

        System.out.println(answer);
    }
}

File: 02 백준/10 문자열/Python/Prob1316.py

"""
title  : 1316. 그룹 단어 체커 (Silver 5)
time   : 40ms        108ms
memory : 31120KB     109240KB
"""

import sys

def solution():

    input = sys.stdin.readline
    N = int(input())                # 단어의 개수
    words = []                      # 단어들
    for _ in range(N):
        words.append(input())

    answer = 0

    for word in words:
        answer += group_word(word)
    
    return answer

def group_word(word):

    check = []
    idx = 0

    while idx < len(word):

        temp = word[idx]

        if temp in check:
            return 0
        
        check.append(temp)
        
        while idx < len(word) and temp == word[idx]:
            idx += 1
    
    return 1

print(solution())